File: src/test/java/com/wxl/dyttcrawler/core/CrawlerTest.java
package com.wxl.dyttcrawler.core;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit.jupiter.SpringExtension;

/**
 * Create by wuxingle on 2020/7/18
 */
@ExtendWith(SpringExtension.class)
@SpringBootTest
public class CrawlerTest {

    @Autowired
    private Crawler crawler;

    @Test
    public void test() throws Exception {
        boolean start = crawler.start(5);
        crawler.awaitStopped();
        crawler.close();
    }
}

File: src/test/kotlin/com/wxl/dyttcrawler/core/CrawlerTest.kt
package com.wxl.dyttcrawler.core

import kotlinx.coroutines.delay
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * Create by wuxingle on 2021/10/12
 *
 */
@ActiveProfiles("redis")
@ExtendWith(SpringExtension::class)
@SpringBootTest
class CrawlerTest {

    @Autowired
    lateinit var crawler: Crawler

    @Test
    fun test() {
        crawler.start()

        runBlocking {
            delay(60 * 1000L)
            crawler.stop()
            crawler.awaitStopped()
        }
    }
}